src/gennylib/src/PoolManager.cpp

```diff
@@ -60,7 +60,11 @@ mongocxx::pool::entry genny::v1::PoolManager::client(const std::string& name,
     std::unique_lock<std::mutex> lock{lap.first};
 
     Pools& pools = lap.second;
-
+    auto perInstance =
+        context["Clients"][name]["PerInstance"].maybe<bool>().value_or(false);
+    if (!perInstance) {
+        instance = 0;
```

Comment:
Isn't this changing the existing behavior as it will overwrite the instance value passed into the function? Did you maybe mean the opposite?

Reply:
No, I may have named it badly but if _perInstance_ is false (the default) then _instance_ will always be 0 so the pool will be shared. If _perinstance_ is true then the _instance_ value will be unqiue and the pool won't be shared.

Maybe a client property called _Shared_  with a default value of true would be clearer?

Also the current code base only ever uses a zero value for this parameter.

Reply:
Ah, the current code base only every using zero was what I was missing.

Reply:
Maybe a comment would help.